### DISCUSSION TITLE
Support Cloudflare Vary for images by adding .jpg extension to /_next/image

### DISCUSSION THREAD

**Discussion by longzheng** [vercel/next.js#39510] [Ideas]
### Describe the feature you'd like to request

The [`next/image` `<Image>` component](https://nextjs.org/docs/basic-features/image-optimization) and default loader automatically optimizes images to WebP (and optionally AVIF) using the loader path `/_next/image?url=...`.

When combined with the Cloudflare CDN, the default behaviour is that WebP variants will be cached and served to any clients even if they don't support WebP because Cloudflare does not adhere to the `Vary` header by default. This breaks some older clients like IE11 which obviously do not support WebP.

[One workaround previously documented](https://st4ng.medium.com/how-to-use-next-js-image-optimization-with-cloudflare-569da7b3ddc6) was to add a special transform rule to Cloudflare that appends a `&webp=` parameter to the path for cache-keying purposes.

Since that workaround was published, Cloudflare [actually announced a feature](https://blog.cloudflare.com/vary-for-images-serve-the-correct-images-to-the-correct-browsers/) called [Vary for Images](https://developers.cloudflare.com/cache/about/vary-for-images/) for this specific scenario of using the `Vary` and `Accept` header to dynamically serve and caching different versions of an image.

However one limitation of Vary for Images is that it requires the URL path must include an [image extension](https://developers.cloudflare.com/cache/about/vary-for-images/#file-extensions) from their list of supported image extensions
```
.avif
.bmp
.gif
.jpg
.jpeg
.jp2
.png
.tif
.tiff
.webp
```

This means Next.js's default `/_next/image` path is not supported natively.

### Describe the solution you'd like

I'm not sure if it makes sense but possibly there could be a `next.config.js` option to add an extension to the default loader path.

For example

```javascript
const nextConfig = {
    images: {
        extension: '.jpg'
    }
}
```

This would change the loader path to `/_next/image.jpg?url=...` so that it is compatible with Cloudflare's Vary for Images.

### Describe alternatives you've considered

I was able to come up with a workaround by using the [`rewrites` function](https://nextjs.org/docs/api-reference/next.config.js/rewrites) and overriding the `images.path` config.

```javascript
const nextConfig = {
    images: {
        path: '/_next/image.jpg', // change the loader path
    },
    async rewrites() {
        return {
            beforeFiles: [
                {
                    source: '/_next/image.jpg', // match the same path above
                    destination: '/_next/image', // route requests to the default loader
                },
            ],
        };
    },
};
```

**Comment by balazsorban44** [comment]
I believe you could achieve this by a custom loader where you can resolve the image url as you want https://nextjs.org/docs/api-reference/next/image#loader

**Reply by longzheng** [reply]
> I believe you could achieve this by a custom loader where you can resolve the image url as you want https://nextjs.org/docs/api-reference/next/image#loader

I'm not sure if changing the loader is necessary since I can already change the default loader path using the `next.config.js` `image.path`.
